js_methods.rb

```diff
@@ -0,0 +1,253 @@
+puts([1, 2, 3, 4, 5].select)
+p 'my_all?'
+puts([2, 4, 6].my_all?(&:even?))
+p 'my_any?'
+puts([2, 4, 6].my_any?(&:odd?))
+p 'my_none?'
+puts([2, 4, 6].my_none?(&:odd?))
+p 'my_count'
+puts([2, 3, 2].my_count(2))
+puts([2, 3, 2].count(2))
+
+my_n_array.each { |e| print "e+2: #{e + 2};" }
+my_n_array.my_each { |e| p "e+2: #{e + 2}" }
+my_n_array.each
+my_n_array.my_each
+p '----- map method -----'
+p(my_n_array.map { |e| e * 2 })
+p(my_n_array.my_map { |e| e * 2 })
+p my_n_array.map
+p my_n_array.my_map
+p '----- my_inject method with array -----'
+p(my_n_array.inject { |a, b| a * b })
+p(my_n_array.my_inject { |a, b| a * b })
+p '----- my_inject method with Symbol -----'
+p my_n_array.inject(1, :*)
+p my_n_array.my_inject(1, :*)
+p(my_n_array.inject(3) { |a, b| a * b })
+p(my_n_array.my_inject(3) { |a, b| a * b })
+p my_n_array.inject(:+)
+p my_n_array.my_inject(:+)
+p '----- multiply_els -----'
+p multiply_els([2, 4, 5])
+p '----- my_map method with Proc -----'
+bproc = proc { |e| p "Test: #{e}" }
+[1, 2, ""].my_map_bproc(&bproc)
+[1, 2, ""].my_map_bproc { |e| p "Test: #{e}" }
+[1, 2, ""].my_map(&bproc)
+[1, 2, ""].my_map { |e| p "Test: #{e}" }
+[1, 2, ""].map(&bproc)
+[1, 2, ""].map { |e| p "Test: #{e}" }
+# .send   , checkITout
+puts '--- .each & my_each with hash ---'
+my_hash.each { |k, v| p "#{k}, #{v}" }
+my_hash.my_each { |k, v| p "#{k}, #{v}" }
+my_hash.each { |k| p k }
+my_hash.my_each { |k| p k }
+puts '--- .each & my_each with array ---'
+my_n_array.each { |k, v| p "#{k}, #{v}" }
+my_n_array.my_each { |k, v| p "#{k}, #{v}" }
+
+p my_hash.all?
+p my_hash.my_all?
+p 'my_all 4 Class'
+p [1, 2, 3].all?(Numeric)
+p [1, 2, 3].my_all?(Numeric)
+p 'my_all 4 Regex'
+p %w[d 2d d].all?(/r/)
+p %w[d 2d d].my_all?(/r/)
+p 'my_all 4 Pattern'
+p [3, 3, 2].all?(3)
+p [3, 3, 2].my_all?(3)
+
+puts
+p 'my_none simple_w_hash'
+p my_hash.none?
+p my_hash.my_none?
+p 'my_none 4 Class'
+p [1, 2, 'b'].none?(Numeric)
+p [1, 2, 'b'].my_none?(Numeric)
+p 'my_none 4 Regex'
+p %w[d 2d d].none?(/r/)
+p %w[d 2d d].my_none?(/r/)
+p 'my_none 4 Pattern'
+p [3, 3, 2].none?(3)
+p [3, 3, 2].my_none?(3)
+
+puts
+p 'my_any simple_w_hash'
+p my_hash.any?
+p my_hash.my_any?
+p 'my_any 4 Class'
+p [1, 2, 'd'].any?(Numeric)
+p [1, 2, 'd'].my_any?(Numeric)
+p 'my_any 4 Regex'
+p %w[d 2d d].any?(/r/)
+p %w[d 2d d].my_any?(/r/)
+p 'my_any 4 Pattern'
+p [3, 3, 2].any?(3)
+p [3, 3, 2].my_any?(3)
+
+p 'my_count'
+puts([2, 3, 2].my_count(2))
+puts([2, 3, 2].count(2))
+puts([2, 3, 2].my_count { |e| (e % 3).zero? })
+puts([2, 3, 2].count { |e| (e % 3).zero? })
+puts([2, 3, 2].my_count)
+puts([2, 3, 2].count)
+
+puts
+p '----- my_inject method with array -----'
+p(my_n_array.inject { |a, b| a * b })
+p(my_n_array.my_inject { |a, b| a * b })
+p '----- my_inject method with Symbol -----'
+p my_n_array.inject(1, :*)
+p my_n_array.my_inject(1, :*)
+p(my_n_array.inject(3) { |a, b| a * b })
+p(my_n_array.my_inject(3) { |a, b| a * b })
+p my_n_array.inject(:+)
+p my_n_array.my_inject(:+)
```

Comment:
:stop_sign: 

Always remember to never push debugging information to public repos. **Please remove!**

Reply:
Roger. Done.